src/PartitionService.ts

```diff
@@ -101,7 +90,79 @@ export class PartitionService {
         return Math.abs(partitionHash) % this.partitionCount;
     }
 
-    getPartitionCount(): number {
+    /**
+     * If partition table is not fetched yet, this method returns zero
+     *
+     * @return the partition count
+     */
+    public getPartitionCount(): number {
         return this.partitionCount;
     }
+
+    /**
+     * Resets the partition table to initial state.
+     */
+    public reset(): void {
+        this.partitionTable.partitions = new Map<number, UUID>();
```

Comment:
nit: can we simplify these three lines to `this.partitionTable = new PartitionTable()`?